weave-js/src/components/PagePanelComponents/Home/Browse3/pages/ChatView/MessagePanel.tsx

```diff
@@ -16,25 +17,62 @@ export const MessagePanel = ({
   isStructuredOutput,
 }: MessagePanelProps) => {
   const isUser = message.role === 'user';
-  const bg = isUser ? 'bg-cactus-300/[0.48]' : 'bg-moon-100';
+  const isTool = message.role === 'tool';
+  const isSystemPrompt = message.role === 'system';
+  const hasToolCalls = !!message.tool_calls && message.tool_calls.length > 0;
+  const hasContent = !!message.content && message.content.length > 0;
+
   return (
-    <div className={classNames('rounded-[8px] px-16 py-8', bg)}>
-      <div style={{fontVariantCaps: 'all-small-caps'}}>{message.role}</div>
-      {message.content && (
-        <div>
-          {_.isString(message.content) ? (
-            <MessagePanelPart
-              value={message.content}
-              isStructuredOutput={isStructuredOutput}
-            />
-          ) : (
-            message.content.map((p, i) => (
-              <MessagePanelPart key={i} value={p} />
-            ))
+    <div className={classNames('flex gap-8', {'mt-32': !isTool})}>
+      <div className="w-40">
+        {!isUser && !isTool && (
+          <Callout
+            size="small"
+            icon="robot-service-member"
+            color="moon"
+            className="h-32 w-32"
+          />
+        )}
+      </div>
+      <div
+        className={classNames('relative overflow-visible py-8', {
+          'border-t border-moon-250': isTool,
+          // System prompt styles, full width with border
+          'w-full rounded-lg border border-moon-250': isSystemPrompt,
+          // Max width for non-system prompts
+          'max-w-3xl': !isSystemPrompt,
+          'w-3/4': isTool || isStructuredOutput || hasToolCalls,
+          // Justify the message to the right if it's a user message, add cactus background
+          'ml-auto bg-cactus-300/[0.24]': isUser,
+          // Justify the message to the left if it's not a user message
+          'mr-auto': !isUser,
+        })}>
+        {isSystemPrompt && (
+          // We only show the role for system prompts
```

Comment:
I want to use these classes for Prompt objects too, where the role is important information to show